packages/grafana-data/src/field/displayProcessor.ts

```diff
@@ -45,10 +45,16 @@ export function getDisplayProcessor(options?: DisplayProcessorOptions): DisplayP
 
   let unit = config.unit;
   let hasDateUnit = unit && (timeFormats[unit] || unit.startsWith('time:'));
+  let showMs = false;
 
   if (field.type === FieldType.time && !hasDateUnit) {
     unit = `dateTimeAsSystem`;
     hasDateUnit = true;
+    if (field.values && field.values.length > 1) {
+      const end = dateTimeParse(field.values.get(field.values.length - 1)).unix();
+      const start = dateTimeParse(field.values.get(0)).unix();
```

Comment:
@nikki-kiga can we avoid parsing here when the field values are not strings?

think you can add an `if (typeof field.values.get(0) === 'string') {...}` to the above, otherwise just subtract.